flink-test-utils-parent/flink-test-utils/pom.xml

```diff
@@ -57,6 +57,14 @@ under the License.
 			<scope>compile</scope>
 		</dependency>
 
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-core</artifactId>
```

Comment:
considering that this is necessary, I assume on of the previous commits isn't compiling?

Reply:
Correct. There was a compilation error in `JMXJobManagerMetricTest` after moving `TestingUtils` into `flink-core`. Providing `flink-core` as an explicit dependency in `flink-test-utils` solved the issue.

Reply:
Was your expectation for me to squash all commits? If not, then please adjust the commit history such that every commit compiles.

Reply:
I left it like that because I thought it's good enough to see the thought process in the commits. I planned to squash it afterwards. But fair enough: I reorganized the commits into less change units.